src/native.rs

```diff
@@ -377,6 +384,13 @@ impl HasContext for Context {
         Ok(NativeBuffer(non_zero_gl_name(buffer)))
     }
 
+    unsafe fn create_buffers(&self) -> Result<Self::Buffer, String> {
```

Comment:
Does `glCreateBuffers` have any benefits over `glGenBuffers` as used in `create_buffer`? If there are benefits then it seems like we could do it automatically in `create_buffer` and fallback when `glCreateBuffers` isn't available

Reply:
most of the patch comes from https://github.com/fendevel/Guide-to-Modern-OpenGL-Functions/blob/master/README.md . 

genBuffer only reserves the buffer name, but doesn't actually initialize the state (the buffer object itself). so, i can't use it with other dsa functions like glNamedBuffer* unless i first bind it to a buffer binding point like array_buffer. 

but create_buffers will initialize the state while reserving the name. and i can start using it, no need to change the global state of bindings.

Reply:
Got it, thanks for clarifying!

In that case, maybe we could have something like `create_named_buffer` and `create_named_texture`? Something like `create_buffer_initialized` might also work, but `named_xyz` at least seems to match a lot of the other DSA functions which could be useful.

This wouldn't match the original OpenGL naming exactly, but would hopefully make the difference more obvious so people don't accidentally call these expecting them to be the regular buffer creation functions (e.g. `createBuffer` in WebGL)